add firebase_element demo

zoechi opened the pull request.

jakemac53 committed:
regen with new custom_element_apigen to get @BehaviorProxy annotations, and make the neon_animated_pages_test pass

jakemac53 committed:
iron-a11y-keys-behavior tests

jakemac53 committed:
iron-control-state and iron-button-state tests

jakemac53 committed:
iron-fit-behavior test

jakemac53 committed:
iron-menu-behavior test

jakemac53 committed:
add iron-menubar-behavior tests

jakemac53 committed:
iron-overlay-behavior tests

jakemac53 committed:
iron-range-behavior test

jakemac53 committed:
iron-resizable-behavior tests

jakemac53 committed:
iron-validatable-behavior tests

jakemac53 committed:
iron-validator-behavior test

jakemac53 committed:
make test runner happy

jakemac53 committed:
added paper-button-behavior test and fixed up some other tests

jakemac53 committed:
paper-inky-focus-behavior test

jakemac53 committed:
paper-dialog-behavior test

jakemac53 committed:
add back reflectable annotation

jakemac53 committed:
remove smoke

zoechigist committed:
tool to generate boilerplate for examples

zoechigist committed:
improve tool/update.dart to add imports to elements and generate boilerplate for imported elements if they don't exist

zoechigist committed:
add firebase_element demo

jakemac53 committed:
update to work with newest polymer version

jakemac53 committed:
bower update, run custom element apigen

jakemac53 committed:
re-apply docs fixes that hadn't been applied yet and regen

jakemac53 committed:
force iron-ajax 1.0.4

jakemac53 committed:
add paper-item-shared-styles

zoechigist committed:
add firebase_element demo

zoechigist committed:
add firebase_element demo

zoechi commented:
Should there be any copyright header added to the files?
Most files are auto-generated from my script anyway so this would'nt cause much work.

zoechigist committed:
add gold_cc_cvc_input demo

zoechigist committed:
add gold_cc_cvc_input demo

jakemac53 commented:
Yes, please add copyright notices if you don't mind. You can copy the ones from the main repo.

jakemac53 commented on demo/.gitignore:
```diff
@@ -0,0 +1,3 @@
+.github_token
```
probably want to add:

.packages
packages
build
.pub

jakemac53 commented on demo/web/firebase_element/app_element.html:
```diff
@@ -0,0 +1,39 @@
+              order-by-child="height"
+              location="https://dinosaur-facts.firebaseio.com/dinosaurs"
+              data="{{dinosaurs}}"></firebase-collection>
+          <template is="dom-repeat" items="[[dinosaurs]]" as="dinosaur">
```
Just curious, is this working properly for  you? I was porting the firebase-collection test and had some issues with a dom-repeat inside a dom-bind.

zoechi replied:
Seems to work, a list of dinosaurs is shown but I created `<app-element>` so I can avoid `<dom-bind>` but I missed this one.

The intention for `<app-element>` also was to be able to additionally integrate all demos in one single application page with a menu to switch between all the demos. 
Polymer.js uses `<iframe>` for this but I don't think this is the optimal solution for Dart.
I just run into an issue with this approach because https://github.com/dart-lang/polymer_elements/blob/master/lib/src/paper-styles/demo-pages.html doesn't work this way. It looks like it needs to be changed from `<style is="custom-style">` to `<dom-element id="some-id">` 
What do you think about how to proceed?

jakemac53 replied:
https://github.com/PolymerElements/paper-styles/blob/master/demo/index.html is an example of how they are using the `demo-pages.html`

zoechi replied:
Yup, but some styles are not applied when I wrap the demo inside an `<app-element>`.

To import styles in the entry page you use `<style is="custom-style"...>` in the imported file, to import the styles in an element the styles need to be wrapped in `<dom-element id="demo-pages"><template>`.
See also: https://www.polymer-project.org/1.0/docs/devguide/styling.html#style-modules
Some styles in `demo-pages.html` as it currently is, are only applied on the entry page and only target elements in the entry page, but not elements inside other elements.
I modified `polymer_elements/lib/src/paper-styles/demo-pages.html` locally from `<style is="custom-style">` to `<dom-module id="demo-pages><template>` and the styles are applied inside `<app-element>`.

Creating a copy of the about 30 lines of styles in `demo-pages.html` file wrapped as `<dom-module id="demo-pages">` (inside `demo-pages.html` or in a new file) would fix this.
What do you think?

jakemac53 replied:
So if we modify `polymer_elements/lib/src/paper-styles/demo-pages.html` it will just get bashed over the next time we update, which I would like to avoid. You could for now just have a custom `demo-pages.html` file in the `demo` project though. Really this change needs to be made upstream, but as it would presumably break a number of JS demos that might be an issue.

zoechi replied:
Mit freundlichen Grüßen

Günter Zöchbauer
guenter@gzoechbauer.com
+43 (699) 10 18 87 15

On Fri, Sep 25, 2015 at 8:16 PM, Jacob MacDonald notifications@github.com
wrote:

> In demo/web/firebase_element/app_element.html
> https://github.com/dart-lang/polymer_elements/pull/63#discussion_r40459576
> :
> 
> > -  <template>
> > -    <style>
> > -      :host {
> > -        display: block;
> > -      }
> > -    </style>
> >   +
> > -    <div class="horizontal center-justified layout">
> > -      <div class="vertical-section">
> > -        <h3>Dinosaurs by Height</h3>
> > -        <template is="dom-bind">
> > -          <firebase-collection
> > -              order-by-child="height"
> > -              location="https://dinosaur-facts.firebaseio.com/dinosaurs"
> > -              data="{{dinosaurs}}"></firebase-collection>
> > -          <template is="dom-repeat" items="[[dinosaurs]]" as="dinosaur">
> 
> So if we modify polymer_elements/lib/src/paper-styles/demo-pages.html it
> will just get bashed over the next time we update, which I would like to
> avoid. You could for now just have a custom demo-pages.html file in the
> demo project though. Really this change needs to be made upstream, but as
> it would presumably break a number of JS demos that might be an issue.
> 
> This is mostly what I did. I called it `demo-elements` because it's for
> elements instead of pages, not sure if this is a good idea. At least you
> can have a look how it worked out.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/dart-lang/polymer_elements/pull/63/files#r40459576.

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,112 @@
+  @eventHandler
+  bool computeChangePasswordDisabled(
+          String email, String password, String newPassword) =>
+      email == null ||
```
nit: format, or if this is what the formatter did then file a bug :)

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,112 @@
+      email == null || email.isEmpty || password == null || password.isEmpty;
+
+  @eventHandler
+  bool computeRemoveUserDisabled(email, password) =>
```
Might as well put the type annotations here as well

zoechi replied:
I created a bug report dart-lang/dart_style#434

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,112 @@
+
+  @eventHandler
+  errorHandler(dom.Event e, var detail) =>
+      message = 'Error: ${detail['message']}';
```
This should be `set(message, 'Error: ${detail['message']}')`, or call notifyPath. You could also define a getter/setter for the property (and call notifyPath in the setter). Ditto in `userSuccessHandler`.

jakemac53 commented on demo/web/firebase_element/x_pretty_json.dart:
```diff
@@ -0,0 +1,21 @@
+      return '';
+    }
+
+    return new JsonEncoder.withIndent('  ').convert(object);
```
I did not know about `withIndent`, nice!

jakemac53 commented:
looks great, just some tiny nits and the copyright notices

jakemac53 commented:
> Most files are auto-generated from my script anyway so this would'nt cause much work.

Curious to know more about this script ;)

zoechi commented:
> Curious to know more about this script ;)

I plan to commit it to the repo but the code is still too ugly and mostly experimental state. I'll port a few more examples from the generated stubs to see if the result is satisfying and then I'll clean up the code and create a PR.
I also included check for changes but couldn't test it, because there where none yet.

zoechigist committed:
- add gold_cc_expiration_input demo
- add gold_cc_input demo
- add styles/demo_elements to make top-level styles work with demos wrapped in `<app-element>`
- add license text in all files
- firebase-element
  - replace `dom-bind` with properties in app-element
  - add missing type annotation
  - use `set('message', ...)`
- add missing entries to `.gitignore`

zoechigist committed:
- add gold_cc_expiration_input demo
- add gold_cc_input demo
- add styles/demo_elements to make top-level styles work with demos wrapped in `<app-element>`
- add license text in all files
- firebase-element
  - replace `dom-bind` with properties in app-element
  - add missing type annotation
  - use `set('message', ...)`
- add missing entries to `.gitignore`

zoechi commented:
The latest update should address all your comments.
I added a `lib/styles/demo_elements.*` element containing the styles from `demo-pages.html`. We could leave it this way or maybe you have a better idea how to deal with this.

jakemac53 commented:
I left some messages on the newest commit, I always forget those don't show up in the PR :/

zoechi commented:
Thanks for the hint with `notify`. Wasn't quite sure how and when to use it. I tried it because the example didn't work as expected, but in the end turned out to be another mistake, just forgot to remove it later.

> I left some messages on the newest commit, I always forget those don't show up in the PR :/

I'm not quite sure how to handle PRs and successive updates. 
For later elements there probably won't as much communication anyway.
I get email notifications at least for your comments on commits (just takes some time until they arrive).

zoechi commented:
> missed this before, but this attribute is missing the quotes around [[dinosaursScores]]. 

Good catch!

zoechigist committed:
- remove `notify: true`
- change `Object` to `var`
- add `"` around `[[..]]`

zoechigist committed:
- remove `notify: true`
- change `Object` to `var`
- add `"` around `[[..]]`

zoechigist committed:
- switch to Polymer 1.0.0-rc.1 branch
- add title tag to index.html files
- add directories to library names to avoid conflicts when several demos are used in the same application

- add gold-cc-expiration-input
- add gold-cc-input
- add gold-email-input
- add gold-phone-input
- add gold-zip-input
- add google-analytics
- add google-apis
- add google-calendar
- add google-castable-video
- add google-chart
- add google-drive
- add google-feeds
- add google-hangout-button
- add google-map

zoechigist committed:
- switch to Polymer 1.0.0-rc.1 branch
- add title tag to index.html files
- add directories to library names to avoid conflicts when several demos are used in the same application

- add gold-cc-expiration-input
- add gold-cc-input
- add gold-email-input
- add gold-phone-input
- add gold-zip-input
- add google-analytics
- add google-apis
- add google-calendar
- add google-castable-video
- add google-chart
- add google-drive
- add google-feeds
- add google-hangout-button
- add google-map

zoechigist committed:
- change `app-element` to `element-name-demo` to be able to use them in the same application
- add google-sheets demo

zoechigist committed:
- change `app-element` to `element-name-demo` to be able to use them in the same application
- add google-sheets demo

zoechigist committed:
- move `packages/browser/dart.js` below the Dart script tag
- add google_signin demo
- add google_streetview_pano demo
- add google_url_shortener demo
- add google_url_youtube demo
- add google_url_youtube_upload demo
- add iron_a11y_announcer demo
- add iron_a11y_keys demo
- add iron_a11y_behavior demo

zoechigist committed:
- move `packages/browser/dart.js` below the Dart script tag
- add google_signin demo
- add google_streetview_pano demo
- add google_url_shortener demo
- add google_url_youtube demo
- add google_url_youtube_upload demo
- add iron_a11y_announcer demo
- add iron_a11y_keys demo
- add iron_a11y_behavior demo

zoechigist committed:
- add boilerplate code generation script `tool/update.dart`
- fix `iron-a11y-keys` (after fixed Polymer issue)
- add iron-ajax demo
- add iron-autogrow-textarea demo
- add iron-behaviors demo
- add iron-checked-element-behavior demo
- add iron-collapse demo
- add iron-component-page demo
- add iron-doc-viewer demo
- add iron-dropdown demo
- add iron-fit-behavior demo
- add iron-flex-layout demo
- add iron-form demo
- add iron-form-element-behavior demo
- add iron-icon demo
- add iron-icons demo
- add iron-iconset demo
- add iron-iconset-svg demo
- add iron-image demo
- add iron-input demo
- add iron-jsonp-library demo
- add iron-list demo

zoechigist committed:
- add boilerplate code generation script `tool/update.dart`
- fix `iron-a11y-keys` (after fixed Polymer issue)
- add iron-ajax demo
- add iron-autogrow-textarea demo
- add iron-behaviors demo
- add iron-checked-element-behavior demo
- add iron-collapse demo
- add iron-component-page demo
- add iron-doc-viewer demo
- add iron-dropdown demo
- add iron-fit-behavior demo
- add iron-flex-layout demo
- add iron-form demo
- add iron-form-element-behavior demo
- add iron-icon demo
- add iron-icons demo
- add iron-iconset demo
- add iron-iconset-svg demo
- add iron-image demo
- add iron-input demo
- add iron-jsonp-library demo
- add iron-list demo

zoechigist committed:
- fix missing mixin in iron_behaviors
- add iron_localstorage demo
- add iron_media_query demo
- add iron_menu_behavior demo

zoechigist committed:
- fix missing mixin in iron_behaviors
- add iron_localstorage demo
- add iron_media_query demo
- add iron_menu_behavior demo

zoechigist committed:
- fix iron-form-element-behavior demo
- add iron-meta demo
- add iron-overlay-behavior demo
- add iron-pages demo
- add iron-range-behavior demo
- add iron-resizable-behavior demo
- add iron-selector demo
- add iron-signals demo
- add iron-validatable-behavior demo

zoechigist committed:
- fix iron-form-element-behavior demo
- add iron-meta demo
- add iron-overlay-behavior demo
- add iron-pages demo
- add iron-range-behavior demo
- add iron-resizable-behavior demo
- add iron-selector demo
- add iron-signals demo
- add iron-validatable-behavior demo

zoechigist committed:
- fix iiron-validatable-behavior demo
- add iron-validator-behavior demo

zoechigist committed:
- fix iiron-validatable-behavior demo
- add iron-validator-behavior demo

jakemac53 committed:
checkpoint

zoechigist committed:
- add all-demos page

- renamed a few helper elements to avoid name collisions in all-demos
- add marked-element demo
- add neon-animation demo
- add paper-badge demo
- add paper-behaviors demo
- add paper-button demo
- add paper-card demo
- add paper-checkbox demo
- add paper-dialog demo
- add paper-dialog-behavior demo
- add paper-dialog-scrollable demo
- add paper-drawer-panel demo
- add paper-dropdown-menu demo
- add paper-fab demo
- add paper-header-panel demo
- add paper-icon-button demo
- add paper-input demo
- add paper-item demo
- add paper-material demo
- add paper-menu demo
- add paper-menu-button demo
- add paper-progress demo
- add paper-radio-button demo
- add paper-radio-group demo
- add paper-ripple demo
- add paper-scroll-header-panel demo
- add paper-slider demo
- add paper-spinner demo
- add paper-styles demo
- add paper-tabs demo
- add paper-toast demo
- add paper-toggle-button demo
- add paper-toolbar demo
- add paper-tooltip demo

zoechigist committed:
- add all-demos page

- renamed a few helper elements to avoid name collisions in all-demos
- add marked-element demo
- add neon-animation demo
- add paper-badge demo
- add paper-behaviors demo
- add paper-button demo
- add paper-card demo
- add paper-checkbox demo
- add paper-dialog demo
- add paper-dialog-behavior demo
- add paper-dialog-scrollable demo
- add paper-drawer-panel demo
- add paper-dropdown-menu demo
- add paper-fab demo
- add paper-header-panel demo
- add paper-icon-button demo
- add paper-input demo
- add paper-item demo
- add paper-material demo
- add paper-menu demo
- add paper-menu-button demo
- add paper-progress demo
- add paper-radio-button demo
- add paper-radio-group demo
- add paper-ripple demo
- add paper-scroll-header-panel demo
- add paper-slider demo
- add paper-spinner demo
- add paper-styles demo
- add paper-tabs demo
- add paper-toast demo
- add paper-toggle-button demo
- add paper-toolbar demo
- add paper-tooltip demo

zoechigist committed:
Merge branch 'master' of github.com:dart-lang/polymer_elements

zoechigist committed:
Merge branch 'demo2' of github.com:bwu-dart-contributing/polymer_elements into demo2

zoechigist committed:
Merge branch 'master' of github.com:dart-lang/polymer_elements into demo2

zoechigist committed:
Merge branch 'master' of github.com:dart-lang/polymer_elements into demo2

jakemac53 commented:
@zoechi looks like I forgot to ever merge this - let me know when things are ready to be looked at again (looks like you have just continued adding more demos which is fine)

jakemac53 committed:
remove  from functions, update firebase_collection_test

zoechi commented:
Its fine, I need to revisit them with the released Polymer version. I hoped for an elements update before that. I saw you updated apigen, which makes it even more likely to get updated elements soon.

jakemac53 committed:
update the gold-* tests

jakemac53 committed:
remove-dep-override

jakemac53 committed:
Merge pull request #81 from dart-lang/gold-tests

update the gold-* tests

jakemac53 committed:
port more tests

zoechigist committed:
Merge branch 'master' of github.com:dart-lang/polymer_elements into demo2

zoechigist committed:
Merge remote-tracking branch 'upstream/update' into demo2

zoechigist committed:
merge upstream

zoechigist committed:
Merge branch 'master' into demo2

zoechigist committed:
undo upstream merge errors

zoechigist committed:
update to published polymer 1.0.0-rc.2 and polymer_elements 1.0.0-rc.2

zoechigist committed:
- fix a few issues in the `update.dart` tool
- add a viewer for the `update_status.json` file (renamed from `update_status.yaml` to `update_status.json`)
- moved all update-related files to tool. Also the `.github_token` file is expected in `tool`

zoechigist committed:
- fix a few issues in the `update.dart` tool
- improve`update_status.json` file viewer
- apply latest changes from JS examples
- update google-map demo to use the new google-map-polys element (not yet available in Dart)
- remove redundant class (SvgSampleIcons) from iron-iconset-svg demo
- add iron-list collapse demo (not yet supported in Dart version)

zoechigist committed:
add documentation about the update tools

zoechigist committed:
minor fixes in the readme

zoechi commented:
@jakemac53 Currently I'm done. Updating the elements would be great, then I'll revisite the demos where I encountered issues.

zoechigist committed:
remove redundant constructor body
break lines
remove commented out code

zoechigist committed:
remove workaround for missing Polymer feature (auto-convert from/to Dart/JS)

jakemac53 commented on demo/pubspec.yaml:
```diff
@@ -0,0 +1,264 @@
+homepage: https://www.dartlang.org/polymer/
+dependencies:
+  browser: any
+  polymer: 1.0.0-rc.2
```
^1.0.0-rc.2

jakemac53 commented on demo/tool/README.md:
```diff
@@ -0,0 +1,215 @@
+
+## Update tool
+
+### </a>Prepare
```
Shouldn't need any explicit links (ditto below). I don't think these work in github markdown?

zoechi replied:
I made this to be able to link to the caption from farther below. I looked it up somewhere and don't know about another way. I tried it by opening the file using the "View" button and it works.

jakemac53 commented on demo/tool/update.dart:
```diff
@@ -0,0 +1,791 @@
+
+/// Creates a library name `web.some_element.some_element_demo` from a path
+/// `web/some_element/some_element_demo`
+String _pathToLibraryName(String path) => '${path.split('/').join('.')}.';
```
extra `.` at the end here it looks like

jakemac53 commented on demo/tool/update_status.dart:
```diff
@@ -0,0 +1,170 @@
+                  .where((file) => file['change'] != 'FileChange.none')
+                  .isNotEmpty);
+    }
+    return repositories.map((repo) => convertToJs(repo)).toList();
```
you shouldn't have to manually use `convertToJs` manually, if you do then its probably a bug

zoechi replied:
It fails at line 80 if I remove it

jakemac53 commented on demo/tool/update_status.dart:
```diff
@@ -0,0 +1,170 @@
+
+  @reflectable
+  Object getRepositoryData(Map data, String organization, Object repository) {
+    return convertToJs(_getRepositoryData(data, organization, repository));
```
ditto

jakemac53 commented on demo/web/all_demos.dart:
```diff
@@ -0,0 +1,404 @@
+
+
+/// Silence analyzer [DemoElements], [PaperItem], [PaperDrawerPanel],
+/// [PaperHeaderPanel], [IronList], [PaperToolbar],
```
probably makes sense to just condense all of these onto as few lines as possible

zoechi replied:
Seems like a lot of work especially when one gets added later somewhere in the middle and all the linebreaks below have to be adjusted.

jakemac53 commented on demo/web/all_demos.dart:
```diff
@@ -0,0 +1,404 @@
+  AllDemos.created() : super.created();
+  List<DemoElementItem> __demos;
+  @property List<DemoElementItem> get demos => __demos;
+  set demos(value) {}
```
remove?

zoechi replied:
Oh, I was fighting with some bug at some point. This is unintended leftover.

jakemac53 commented on demo/web/all_demos.dart:
```diff
@@ -0,0 +1,404 @@
+@PolymerRegister('all-demos')
+class AllDemos extends PolymerElement {
+  AllDemos.created() : super.created();
+  List<DemoElementItem> __demos;
```
any particular reason for the double underscore?

zoechi replied:
This was a backing field for a getter/setter with a private setter, therefore the name `_emos` was already taken. Can't remember exactly why I did this. Just a leftover I forgot to remove.

jakemac53 commented on demo/web/all_demos.dart:
```diff
@@ -0,0 +1,404 @@
+
+  @reflectable
+  void demoClickHandler(dom.Event event, [_]) {
+    final item = convertToDart(_demoList.jsElement.callMethod(
```
Mind referencing https://github.com/dart-lang/polymer_elements/issues/90 in a comment above this?

jakemac53 commented on demo/web/firebase_element/index.html:
```diff
@@ -0,0 +1,27 @@
+  <body unresolved>
+    <firebase-element-demo></firebase-element-demo>
+
+    <script src="packages/browser/dart.js"></script>
```
I think this has to come after the dart script tag

zoechi replied:
Right, good catch :+1:

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    //_firebaseLogin.login(parameters, null);
+    _firebaseLogin.jsElement.callMethod('login', [parameters, null]);
```
please mention https://github.com/dart-lang/polymer_elements/issues/59 here

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,122 @@
+  void logout([_, __]) => _firebaseLogin.logout();
+
+  @reflectable
+  errorHandler(dom.Event e, var detail) =>
```
no need for `var`

zoechi replied:
variables, parameters and functions without type annotations always look like broken/experimental code to me. Removed anyway :)

jakemac53 commented on demo/web/firebase_element/x_login.dart:
```diff
@@ -0,0 +1,122 @@
+  @reflectable
+  bool computeChangePasswordDisabled(
+          String email, String password, String newPassword) =>
+      email == null ||
```
formatter bug?

zoechi replied:
I think this is correct / as intended. 
After `=>` the indentation is reduced to be relative to the start of the parent expression and after `email == null ||` indentation is again increased to indicate that it continues the previous line.
Actually, I don't feel entitled to discuss the results of the formatter. I'm just glad I don't have to make all these decisions myself :sweat:

jakemac53 replied:
Haha fair enough, I certainly am not suggesting that this code should be changed from what the formatter does. At the most I would just file a bug with the formatter if we thought that this was "wrong".

jakemac53 commented:
@zoechi have you used reitveld before? For a cl of this size it would help me a lot when reviewing (and its also easier to see changes/responses over time).

jakemac53 commented on demo/web/google_castable_video/google_castable_video_demo.dart:
```diff
@@ -0,0 +1,68 @@
+  //listen for casting event to change icon
+  @reflectable
+  casting(dom.CustomEvent event, [_]) {
+    ($['cast'] as dom.ButtonElement).innerHtml =
```
Seems like this should just be a property and a binding? If it matches the JS demo I guess that is ok though.

zoechi replied:
This is actually a response to 

> @zoechi have you used rietveld before? For a cl of this size it would help me a lot when reviewing (and its also easier to see changes/responses over time). It is also to big for the github UI to handle lol.

because this comment of yours has no reply button.
I haven't used Rietveld before but if you can point out where to start I'd give it a try.

zoechi replied:
> Seems like this should just be a property and a binding? If it matches the JS demo I guess that is ok though.

Right. I usually changed it when I encountered something like that in the JS demos. This was an oversight.

The JS demos are often not quite polymery. Sometimes they don't even use `dom-bind`.
Moving them into components "forced" me to improve them. IMHO the demos should not only demonstrate how the components look like but also how they are being used in a Polymer application. Not every Polymer element is used on the entry page :-/ </ranting>

jakemac53 commented on demo/web/google_drive/google_drive_demo.dart:
```diff
@@ -0,0 +1,49 @@
+
+  @reflectable
+  void uploadStarted(dom.CustomEvent event, [_]) {
+    set('queueUploadList', (event.detail.files as List).toList());
```
isn't `event.detail` a `JsObject` or a `Map` (ditto below).

zoechi replied:
There is no `google-drive` component yet in dart-lang/polymer_elements. This code is not tested.

jakemac53 replied:
ah ok that would explain it ;)

zoechigist committed:
Fixes @jakemac53 s review comments
- pubspec.yaml
  - fix Polymer version constraint (add `^`)
- firebase_element
  - move `browser/dart.js` script tag below `index.dart`
  - remove 'var' parameter type annotation
- google_castable_video
  - make button caption a binding to a property instead of `innerHtml = `
- all_demos
  - remove leftover from backing field with private setter

jakemac53 commented on demo/web/iron_a11y_keys/x_key_aware.dart:
```diff
@@ -0,0 +1,43 @@
+
+  String __pressed = '';
+  @property String get pressed => __pressed;
+  set _pressed(String value) {
```
any particular reason to make the setter private? (ditto for some other elements below)

zoechi replied:
In the JS example the property is read-only and I just imitated it

``` js
 properties: {
      pressed: {
        type: String,
        readOnly: true,
        value: ''
      },
```

jakemac53 replied:
ah ok sure that makes sense then

jakemac53 commented:
Sorry for all the flip-flopping, but the more that I look at this the more I am thinking that we should actually go back to the old way of having a separate repo for the demos.

My primary concern is maintenance of the demos. This is a lot of code which is going to have a high churn rate, and making it be part of the primary package adds a major barrier to updating the elements in a timely fashion. If these live in a separate repo then they can be updated on a different timeline without slowing down the overall updating process.

zoechi commented:
No prob. It's an experiment. I'm still trying to figure what strategy with examples works best in what situation (for example also in bwu_datagrid). Please create a repo for the demos and I create a new PR.

jakemac53 commented:
@zoechi I think it actually makes sense to have this as a repo under your personal account, and we will link to it from the readme? That way you can maintain credit for it and also keep it up to date without having to go through code reviews from us etc.

zoechi commented:
Actually I like your review comments :) Ok,  I'll create a repository. If you want to migrate it somewhere else at some point just reach out.

jakemac53 commented:
Sure, and feel free to keep sending me code reviews, I am happy to look at them :)

jakemac53 commented:
closing for now, let me know once these are available elsewhere and I will link on the README, or just send a PR to update the README. Thanks!

zoechi commented:
Yup, well do. Thanks for the support so far!
